packages/puppeteer-core/src/bidi/core/Realm.ts

```diff
@@ -0,0 +1,256 @@
+import type * as Bidi from 'chromium-bidi/lib/cjs/protocol/protocol.js';
+
+import {EventEmitter} from '../../common/EventEmitter.js';
+
+import type {BrowsingContext} from './BrowsingContext.js';
+import type {Connection} from './Connection.js';
+
+export type CallFunctionOptions = Omit<
+  Bidi.Script.CallFunctionParameters,
+  'functionDeclaration' | 'awaitPromise' | 'target'
+>;
+
+export type EvaluateOptions = Omit<
+  Bidi.Script.EvaluateParameters,
+  'expression' | 'awaitPromise' | 'target'
+>;
+
+export abstract class Realm extends EventEmitter<{
+  /** Emitted when the realm is destroyed. */
+  destroyed: void;
+  /** Emitted when a dedicated worker is created in the realm. */
+  worker: DedicatedWorkerRealm;
+  /** Emitted when a shared worker is created in the realm. */
+  sharedworker: SharedWorkerRealm;
+}> {
+  readonly id: string;
+  readonly origin: string;
+  protected constructor(id: string, origin: string) {
+    super();
+    this.id = id;
+    this.origin = origin;
+  }
+
+  protected initialize(): void {
+    this.connection.on('script.realmDestroyed', info => {
+      if (info.realm === this.id) {
+        this.emit('destroyed', undefined);
+      }
+    });
+  }
+
+  protected abstract get connection(): Connection;
+
+  protected get target(): Bidi.Script.Target {
+    return {realm: this.id};
+  }
+
+  async disown(handles: string[]): Promise<void> {
+    await this.connection.send('script.disown', {
+      target: this.target,
+      handles,
+    });
+  }
+
+  async callFunction(
+    functionDeclaration: string,
+    awaitPromise: boolean,
+    options: CallFunctionOptions = {}
+  ): Promise<Bidi.Script.EvaluateResult> {
+    const {result} = await this.connection.send('script.callFunction', {
+      functionDeclaration,
+      awaitPromise,
+      target: this.target,
+      ...options,
+    });
+    return result;
+  }
+
+  async evaluate(
+    expression: string,
+    awaitPromise: boolean,
```

Comment:
Can you point me to a the code where we use `awaitPromise: false`?

Reply:
We just want to keep it here because that's a required argument in `script.evaluate`. Required args are parametrized while optional ones are put into an object.